testsuite/features/reposync/reference_srv_sync_products_extra.feature

```diff
@@ -37,7 +37,7 @@ Feature: Synchronize extra products in the products page of the Setup Wizard
     When I follow the left menu "Admin > Setup Wizard > Products"
     And I wait until I do not see "Loading" text
     And I enter "Ubuntu" as the filtered product description
-    And I select "amd64-deb" in the dropdown list of the architecture filter
+    And I select "amd64-deb" from "product-arch-filter"
```

Comment:
Ideally, we better hide this technical detail in the step definition.
That way, if we change this id in the future, we only need to change it in the step definition and not in all the scenarios using it.

Reply:
Hi @Etheryte the Gherkin syntax and Cucumber scenarios have been designed to use natural language to describe the steps a user must follow to achieve something. Ideally, we should write the scenarios as if were on the official documentation. So, even if other steps don't follow this good practice, let's try to do it properly wherever we can.

The good news is that's easy to fix, we only need to create a new step definition, where inside it we use this id.

Reply:
Yes, we should avoid using internals as much as we could. The "feature" part should remain "as the end user sees it", and end users don't see ids when they are in front of the UI, they see texts...

That being said, instead of adding new specialized steps, we might try to use this existing one:
```
When(/^I select "(.*?)" from "([^"]*)" dropdown/) do |selection, label|
```
This one uses a label instead of an id. However, it assumes the label is in "name" attribute, so I am not sure it would work. To be investigated.